pkg/controllers/dashboard/kubernetesprovider/kubernetesprovider.go

```diff
@@ -33,8 +34,10 @@ func Register(ctx context.Context,
 
 func (h *handler) OnChange(key string, cluster *v3.Cluster) (*v3.Cluster, error) {
 	if cluster == nil || cluster.Status.Provider != "" {
-		// if cluster has windows enabled and provider is not rke.windows, continue to detect if it has windows nodes
-		if cluster == nil || !cluster.Spec.WindowsPreferedCluster || cluster.Status.Provider == "rke.windows" {
+		// rke1 and rke2 windows cluster check
+		// rke1: if cluster has windows enabled and provider is not rke.windows, continue to detect if it has windows nodes
+		// rke2: if provider is not rke2.windows, continue to detect if it has windows nodes
+		if cluster == nil || !cluster.Spec.WindowsPreferedCluster || strings.HasSuffix(cluster.Status.Provider, ".windows") {
```

Comment:
Actually...Do we set `WindowsPreferedCluster` for rke2? Otherwise this will not work

Reply:
We don't set that for rke2 clusters. do I need to have a second check specifically for rke2 that doesn't check `WindowsPreferedCluster`? something like this:
```
if cluster == nil || cluster.Status.Provider == "rke2.windows" {
```

Reply:
Yeah we need similar logic for that. For rke1 use the existing logic. For rke2 we just have to constantly process it, because at any time if a window node is registered then we treat it as windows cluster. Feel bad that this is getting more complicated.

Reply:
Is there a way to know if a rke2 cluster is going to a windows cluster? any field on api?

Reply:
A RKE2 cluster can be a linux cluster *until* you add a windows worker node. It's unlike RKE1 where you define it up front.